components/server/src/database/dsls/object_dsl.rs

```diff
@@ -162,10 +172,33 @@ impl CrudDb for Object {
     async fn get(id: impl PrimaryKey, client: &Client) -> Result<Option<Self>> {
         let query = "SELECT * FROM objects WHERE id = $1";
         let prepared = client.prepare(query).await?;
-        Ok(client
-            .query_opt(&prepared, &[&id])
-            .await?
-            .map(|e| Object::from_row(&e)))
+        let mut backoff_counter: u64 = 0;
+
+        while backoff_counter <= *MAX_RETRIES {
+            match client.query_opt(&prepared, &[&id]).await {
+                Ok(query) => return Ok(query.map(|e| Object::from_row(&e))),
+                Err(err) => {
+                    if let Some(sql_error) = err.code() {
+                        match sql_error.code() {
+                            // => 40001 Error from yugabyte matches to 25P02
+                            "25P02" => {
+                                backoff_counter += 1;
+                                tokio::time::sleep(Duration::from_millis(
+                                    *RETRY_TIMEOUT ^ backoff_counter,
+                                ))
+                                .await;
+                            }
+                            _code => {
+                                return Err(anyhow!(err));
+                            }
+                        }
+                    } else {
+                        return Err(anyhow!(err));
+                    }
+                }
+            }
+        }
+        Err(anyhow!("TODO"))
```

Comment:
We should package the re-try functionality into a reusable form in the foreseeable future, such as a macro or a decorator.

Reply:
This is a good idea, especially if the retry logic is extended for more database calls